src/send/usePrepareSendTransactions.ts

```diff
@@ -0,0 +1,108 @@
+import { useState } from 'react'
+import {
+  getMaxGasCost,
+  PreparedTransactionsResult,
+  prepareERC20TransferTransaction,
+} from 'src/viem/prepareTransactions'
+import { TokenBalance, tokenBalanceHasAddress } from 'src/tokens/slice'
+import BigNumber from 'bignumber.js'
+import { useAsyncCallback } from 'react-async-hook'
+import { tokenSupportsComments } from 'src/tokens/utils'
+import Logger from 'src/utils/Logger'
+import { tokenAmountInSmallestUnit } from 'src/tokens/saga'
+
+const TAG = 'src/send/usePrepareSendTransactions'
+
+// just exported for testing
+export function _getOnSuccessCallback({
+  setFeeCurrency,
+  setPrepareTransactionsResult,
+  setFeeAmount,
+}: {
+  setFeeCurrency: (token: TokenBalance | undefined) => void
+  setPrepareTransactionsResult: (result: PreparedTransactionsResult | undefined) => void
+  setFeeAmount: (amount: BigNumber | undefined) => void
+}) {
+  return (result: PreparedTransactionsResult | undefined) => {
+    setPrepareTransactionsResult(result)
+    if (result?.type === 'possible') {
+      setFeeCurrency(result.feeCurrency)
+      setFeeAmount(
+        getMaxGasCost(result.transactions).dividedBy(
+          new BigNumber(10).exponentiatedBy(result.feeCurrency.decimals)
+        )
+      )
+    } else {
+      setFeeCurrency(undefined)
+      setFeeAmount(undefined)
+    }
+  }
+}
+
+// just exported for testing
+export async function _prepareSendTransactionsCallback({
+  amount,
+  token,
+  recipientAddress,
+  walletAddress,
+  isDekRegistered,
+  feeCurrencies,
+}: {
+  amount: BigNumber
+  token: TokenBalance
+  recipientAddress: string
+  walletAddress: string
+  isDekRegistered: boolean
+  feeCurrencies: TokenBalance[]
+}) {
+  const includeRegisterDekTx = !isDekRegistered && tokenSupportsComments(token)
+
+  if (amount.isLessThanOrEqualTo(0)) {
+    return
+  }
+  if (tokenBalanceHasAddress(token)) {
+    if (!includeRegisterDekTx) {
+      return prepareERC20TransferTransaction({
+        fromWalletAddress: walletAddress,
+        toWalletAddress: recipientAddress,
+        sendToken: token,
+        amount: BigInt(tokenAmountInSmallestUnit(amount, token.decimals)),
+        feeCurrencies,
+      })
+    }
+    // TODO(ACT-955): case where we need to register DEK too
+  }
+  // TODO(ACT-956): non-ERC20 native asset case
+}
+
+export function usePrepareSendTransactions(
+  prepareSendTransactionsCallback = _prepareSendTransactionsCallback // just for testing
+) {
+  const [prepareTransactionsResult, setPrepareTransactionsResult] = useState<
+    PreparedTransactionsResult | undefined
+  >()
+  const [feeCurrency, setFeeCurrency] = useState<TokenBalance | undefined>()
+
+  const [feeAmount, setFeeAmount] = useState<BigNumber | undefined>()
+  const prepareTransactions = useAsyncCallback(prepareSendTransactionsCallback, {
+    onError: (error) => {
+      Logger.error(TAG, `prepareTransactionsOutput: ${error}`)
```

Comment:
I noticed this error is thrown with fee section just showing a spinner when I choose CELO and enter an amount close to max. I think we'd want to pick another fee currency and return tx is possible if there's balance for them (I did have cUSD, cEUR balance) or return a max-amount response? But maybe this is out of scope for this PR since it seems like an issue with the `prepareERC20TransferTransaction` function

Reply:
yes I noticed that too. I want to handle it separately.